app/dts/behaviors/leader_key.dtsi

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+ / {
+	behaviors {
+		/omit-if-no-ref/ leader: leader_key {
```

Comment:
It seems to me we shouldn't limit this to just one global `leader` key, as this limits folks to just one leader key, and it's associated sequences.

Is there really no use case for different leader keys, with different possible sequences?

Seems like you could have child props in the yaml binding, and something like:

```
leader_foo: leader_foo {
  compatible = "zmk,behavior-leader";
  ...
  seq1 {
  };
  seq2 {
  };
};
````

That would give us a bit more flexibility.

If we really want to make it easy, we could still define one global leader you could enhance with children, e.g.:

```
&leader {
  seq1 {
  };
};
```

Reply:
Right now I use multiple leader keys with layers (see [here](https://github.com/nickconway/zmk-config/blob/main/config/include/leader.dtsi)). I use macros to switch layers before pressing the leader key which has the same effect although is a bit more work for the user.